### PR TITLE
WIP: Battle Module

### LINKED ISSUE #3: Implementar módulo de batalha.

### PR DESCRIPTION
**Pull request by juliobguedes**
Closes #3, #8 . Work in Progress

### CODE REVIEW

File: battle.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+int hero_attacks(hero_in_battle hero, monster enemy) {
+    int action;
+
+    cout << "Choose your action:" << endl;
+    while (1) {
+        action = get_battle_action();
+        if (action == 1) {
+            enemy.health -= hero.hero.strength;
```
Comment (raonismaneoto) [inline battle.cpp]: hero.hero sounds weird

File: battle.cpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    cout << endl;
+}
+
+int hero_attacks(hero_in_battle hero, monster enemy) {
+    int action;
+
+    cout << "Choose your action:" << endl;
+    while (1) {
+        action = get_battle_action();
+        if (action == 1) {
```
Comment (raonismaneoto) [inline battle.cpp]: couldn't you use a switch case statement? I think it would be better

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

File: battle.cpp (hunk 2 above)
Comment (joseims) [reply battle.cpp]: I actually like if and elses more than switch lets discuss it in the slack

File: battle.cpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            enemy.health -= hero.hero.strength;
+            return 1;
```
Comment (joseims) [inline battle.cpp]: Is the hero even doing anything when atacking?

File: battle.cpp [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        } else if (action == 2) {
+            return 0;
```
Comment (joseims) [inline battle.cpp]: Or blocking?

**Review by joseims — CHANGES_REQUESTED** [review]
IS this really supposeed to be a pull request? Cause you have WIP in the title :v

**Comment by joseims** [comment]
Do the merge

File: battle.cpp [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+void initial_message(main_character hero) {
+    cout << "THE BATTLE IS ABOUT TO BEGIN!!" << endl;
+    cout << "Remember Hero, your status is: " << endl;
+    cout << "Attack: " << hero.strength << endl;
+    cout << "Defense: " << hero.defense << endl;
+    cout << "Health: " << hero.health << "/" << hero.max_health << endl;
+    cout << "You have " << hero.potion << " potions. Use them wisely!" << endl << endl;
+}
+
+void battle_workflow(main_character hero, monster enemy) {
```
Comment (joseims) [inline battle.cpp]: I think this method should return a int ( 1 or 0 ) for whenever he won the fight(1) , or he lost(0)

**Pull request merged**